rentacar/src/main/java/ro/sci/carrental/repository/TransactionRepository.java

```diff
@@ -0,0 +1,62 @@
+package ro.sci.carrental.repository;
+
+import ro.sci.carrental.model.calendar.Transaction;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Create a transaction repository object blueprint
+ * Created by tudor on 14.06.2017.
+ */
+public class TransactionRepository implements TransactionRepositoryInterface {
+    private Set<Transaction> transactions;
+
+    public TransactionRepository() {
+        this.transactions = new HashSet<>();
+    }
+
+    public List<Transaction> findTransactionById(int id) {
+        List<Transaction> foundTransactions = new ArrayList<>();
+
+        for (Transaction transaction : transactions) {
```

Comment:
this for loop does unnecessary computation after the transaction is found. Cosinder using some other structure